protocol.go

```diff
@@ -119,6 +119,22 @@ func (p *Conn) Read(b []byte) (int, error) {
 	return p.bufReader.Read(b)
 }
 
+func (p *Conn) ReadFrom(r io.Reader) (int64, error) {
+	if rf, ok := p.conn.(io.ReaderFrom); ok {
+		return rf.ReadFrom(r)
+	}
+	return io.Copy(p.conn, r)
+}
+
+func (p *Conn) WriteTo(w io.Writer) (int64, error) {
+	var err error
+	p.once.Do(func() { err = p.checkPrefix() })
```

Comment:
I think you meant for this line to be in ReadFrom? The Read method has it, but the Write method doesn't. Or is there a specific reason it's here?

Reply:
No don't think so. `ReadFrom` does not read from `Conn`, it reads from the reader that is passed as an argument. It only writes to the `Conn`, in this direction the buffered reader is not involved and the check is not required.

Reply:
Ah, duh. Yes, I didn't pay close enough attention to the direction here. Looks fine!